### PR TITLE
implemented mini-batch extraction with with statement for PyTorch

### PR DESCRIPTION
**Pull request by LukasMut**
- implemented mini-batch extraction using a `with` statement to add hooks at the start of extraction and remove them in the end (this is only relevant for  PyTorch models)

### CODE REVIEW

File: thingsvision/core/extraction/torch.py [hunk 1]
```diff
@@ -230,3 +232,25 @@ def get_default_transformation(
...
+    def __init__(
+        self, extractor: PyTorchExtractor, module_name: str, output_type: str
+    ) -> None:
+        self.extractor = extractor
+        self.module_name = module_name
+        self.output_type = output_type
+
+    def __enter__(self) -> PyTorchExtractor:
+        self.extractor._module_and_output_check(self.module_name, self.output_type)
+        self.extractor.register_hook(self.module_name)
```
Comment (lciernik) [inline thingsvision/core/extraction/torch.py:248]: Minor suggestion: `self.extractor.register_hook` and `self.extractor._unregister_hook()` either both private or public object methods.

**Review by lciernik — COMMENTED** [review]
Added small suggestion.

File: thingsvision/core/extraction/torch.py (hunk 1 above)
Comment (LukasMut) [reply thingsvision/core/extraction/torch.py]: Both are private methods now.

**Review by lciernik — APPROVED** [review]

**Review by a1247418 — APPROVED** [review]
Looks good.

**Pull request merged**